language-plutus-core/common/Common.hs

```diff
@@ -19,32 +19,32 @@ import           System.FilePath            ((</>))
 import           Test.Tasty
 import           Test.Tasty.Golden
 
--- | A 'TestTree' of golden tests stored in some folder.
-type TestNestedGolden = Reader [String] TestTree
+-- | A 'TestTree' of tests stored in some folder.
+type TestNested = Reader [String] TestTree
 
--- | Run a 'TestTree' of golden tests against a folder.
-runTestNestedGoldenIn :: [String] -> TestNestedGolden -> TestTree
-runTestNestedGoldenIn path test = runReader test path
+-- | Run a 'TestTree' of tests against a folder.
+runTestNestedIn :: [String] -> TestNested -> TestTree
+runTestNestedIn path test = runReader test path
 
--- | Descend into a folder of golden tests.
-testNestedGolden :: String -> [TestNestedGolden] -> TestNestedGolden
-testNestedGolden folderName =
+-- | Descend into a folder of tests.
+testNested :: String -> [TestNested] -> TestNested
+testNested folderName =
     Reader.local (++ [folderName]) . fmap (testGroup folderName) . sequence
 
 -- | Check the contents of a file against a 'Text'.
-goldenVsText :: TestName -> FilePath -> Text -> TestTree
-goldenVsText name ref = goldenVsString name ref . pure . BSL.fromStrict . encodeUtf8
+goldenVsText :: TestName -> FilePath -> IO Text -> TestTree
```

Comment:
I'd prefer if we keep `goldenVsText ` that receive sa `Text` and define `goldenVsTextM` that receives a `IO Text`. The pure case is much more common and `goldenVsText name ref text` is a much nicer call than `goldenVsText name ref (pure text)`.

Reply:
If you like - but I think exposing the full underlying API is nice. And the changes required by this are in practice very modest.

Reply:
Yeah, but the changes do not look like improvements to me and we will have more tests of this kind.

Reply:
Done. I think the result is worse - now we just have lots of extra functions which just compose with `pure` but I don't care that much.